internal/resource/key_test.go

```diff
@@ -0,0 +1,246 @@
+					},
+				},
+			},
+		},
+		{
+			name:    "need some content",
+			Key:     goodKey,
+			wantErr: true,
+			Errs: []*v1.ErrorDetail{
+				{
+					Type: v1.ErrorType_ERROR_TYPE_ENTITY,
+					Messages: []string{
+						"Keys must be defined either in JWK or PEM format",
+					},
+				},
+			},
+		},
+		{
+			name: "key with jwk and without kid isn't valid",
+			Key: func() Key {
+				k := goodKey()
+				k.Key.Jwk = &v1.JwkKey{}
+				return k
+			},
+			wantErr: true,
+			Errs: []*v1.ErrorDetail{
+				{
+					Type:  v1.ErrorType_ERROR_TYPE_FIELD,
+					Field: "jwk",
+					Messages: []string{
+						"missing properties: 'kid'",
+					},
+				},
+			},
+		},
+		{
+			name: "key with valid jwk is valid",
+			Key: func() Key {
+				k := NewKey()
+				k.Key.Jwk = &v1.JwkKey{}
+				k.ProcessDefaults(context.Background())
+				return k
+			},
+		},
+		{
+			name: "key with invalid pem is not valid",
+			Key: func() Key {
+				k := NewKey()
+				k.Key.Pem = &v1.PemKey{
+					PublicKey:  "1234",
+					PrivateKey: "hunter2",
+				}
+				k.ProcessDefaults(context.Background())
+				return k
+			},
+			wantErr: true,
+			Errs: []*v1.ErrorDetail{
+				{
+					Type:     v1.ErrorType_ERROR_TYPE_FIELD,
+					Field:    "pem.private_key",
+					Messages: []string{"'hunter2' is not valid 'pem-encoded-private-key'"},
+				},
+				{
+					Type:     v1.ErrorType_ERROR_TYPE_FIELD,
+					Field:    "pem.public_key",
+					Messages: []string{"'1234' is not valid 'pem-encoded-public-key'"},
+				},
+			},
+		},
+		{
+			name: "key with valid pem is valid",
+			Key: func() Key {
+				k := NewKey()
+				k.Key.Pem = &v1.PemKey{
+					PublicKey:  pemTestPublicKey,
+					PrivateKey: pemTestPrivateKey,
+				}
+				k.ProcessDefaults(context.Background())
+				return k
+			},
+		},
+		{
+			name: "key with both jwk and pem is not valid",
+			Key: func() Key {
+				k := NewKey()
+				k.Key.Jwk = &v1.JwkKey{}
+				k.Key.Pem = &v1.PemKey{
+					PublicKey:  pemTestPublicKey,
+					PrivateKey: pemTestPrivateKey,
+				}
+				k.ProcessDefaults(context.Background())
+				return k
+			},
+			wantErr: true,
+			Errs: []*v1.ErrorDetail{
+				{
+					Type:     v1.ErrorType_ERROR_TYPE_ENTITY,
+					Messages: []string{"Keys must be defined either in JWK or PEM format"},
+				},
+			},
+		},
+		{
+			name: "repeated tags isn't valid",
+			Key: func() Key {
+				k := NewKey()
+				k.Key.Jwk = &v1.JwkKey{}
+				k.Key.Tags = []string{"A1", "X3", "A1"}
+				k.ProcessDefaults(context.Background())
+				return k
+			},
+			wantErr: true,
+			Errs: []*v1.ErrorDetail{
+				{
+					Type:     v1.ErrorType_ERROR_TYPE_FIELD,
+					Field:    "tags",
+					Messages: []string{"items at index 0 and 2 are equal"},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			k := tt.Key()
+			err := k.Validate(context.Background())
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Validate() error = %v, wantErr %v", err, tt.wantErr)
```

Comment:
nit: The mixed usages of `assert` and stdlib testing utils is strange, I generally try to stick to one or the other.